app/controllers/api/jobs_controller.rb

```diff
@@ -16,10 +25,53 @@ def info_for_output(jobs)
         url: job["url"],
         latitude: job["latitude"],
         longitude: job["longitude"],
-        key: job["jobkey"],
+        job_key: job["jobkey"],
         location: job["formattedLocationFull"],
         date_posted: job["formattedRelativeTime"]
       }
     end
   end
+
+  def job_params
+    params.require(:job).permit(:jobkeys)
+  end
+
+  def authorize!
+    unless current_user
+      render json: {message: "Please log in", status: :unauthorized}
+    end
+  end
+
+  def info_to_create_job(job)
+    {
+      job_key: job["jobkey"],
+      longitude: job["longitude"],
+      latitude: job["latitude"],
+      company: job["company"],
+      job_title: job["jobtitle"],
+      location: job["formattedLocationFull"]
+    }
+  end
+
+  def create_new_job_and_save(job_key)
```

Comment:
the job model? how? Its created a new job so there is not job object to call this function on then right?